Wobble/Scheduling/TaskHandler.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Wobble.Logging;
+
+namespace Quaver.Shared.Scheduling
+{
+    /// <summary>
+    ///     This will be used to handle threaded tasks.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <typeparam name="TResult"></typeparam>
+    public class TaskHandler<T, TResult> : IDisposable
+    {
+        private Func<T, TResult> Function { get; }
```

Comment:
I think passing the cancellation token through is a good idea so underlying code can also do cancellable operations.